1. Commit

Ferias opened the pull request.

Ferias committed:
1. Commit

Ferias committed:
Testfall mit epocheconverter V1.1

BenediktSt commented on angular/angular.json:
```diff
@@ -0,0 +1,135 @@
+{
```
Diese Datei sollte eigentlich meines Wissens nicht notwendig sein, im Testing-Repo

Ferias replied:
beseitigt

BenediktSt commented on angular/karma.conf.js:
```diff
@@ -0,0 +1,37 @@
+// Karma configuration file, see link for more information
```
Karma wird nur für die Angular Unit tests verwendet oder nicht? Warum brauchen wir die hier?

Ferias replied:
beseitigt

BenediktSt commented on angular/package.json:
```diff
@@ -0,0 +1,49 @@
+    "e2e": "ng e2e"
+  },
+  "private": true,
+  "dependencies": {
```
die ganzen angular dependencies sind nicht notwendig. Bitte überprüfen, was hier wirklich benötigt wird

BenediktSt replied:
ich denke die restlichen dependencies (nicht dev-dependencies) können ebenfalls noch raus

BenediktSt commented on angular/conf.js:
```diff
@@ -0,0 +1,5 @@
+exports.config = {
```
Wofür brauchen wir diese config? sollte Protractor.conf übernehmen

Ferias replied:
Ok von conf.js zu protractor.conf.js umgezogen, habe bisher nur conf.js benutzt

BenediktSt commented on angular/proxy.conf.json:
```diff
@@ -0,0 +1,8 @@
+{
```
nicht notwendig

BenediktSt commented on angular/spec.js:
```diff
@@ -0,0 +1,68 @@
+describe('Protractor Demo App', function() {
```
Datei ebenfalls nicht notwendig

Ferias replied:
Da drin waren meine Übungen, ist jetzt weg

BenediktSt commented on angular/spec2.js:
```diff
@@ -0,0 +1,63 @@
+	var timestamp_hours = element(by.name("hh"));
+	var timestamp_minutes = element(by.name("mn"));
+	var timestamp = element(by.name("hf-result"));
+	var timestamp_button = element.all(by.buttonText("uman date to Timestamp"));
```
uman?

BenediktSt replied:
außerdem verwendest du hier element.all, ist das so gewollt? Sind hier mehrere Buttons?
wenn ja variable so benennen, dass dies auch hervorgeht

Ferias replied:
Existiert nicht mehr

BenediktSt commented on angular/src/main.ts:
```diff
@@ -0,0 +1,12 @@
+import { enableProdMode } from '@angular/core';
```
Bitte angular Dateien entfernen

Ferias committed:
Testfall mit epocheconverter V1.2
- Button Problem gelöst (nicht optimal)
- Input-Felder können manipuliert werden

Ferias committed:
Testfall mit epocheconverter V1.3
- Testfälle erweitert

Ferias committed:
PR angepasst

Ferias committed:
Dependencies angepasst
- dependencies stark reduziert
- devDependencies stark reduziert

BenediktSt commented:
Fast alle Deien liegen noch in einem angular Ordner, das sollte auch noch geändert werden

Ferias committed:
Testfall mit epocheconverter V1.4
- moment.js integrieren
- 1 zusätzlich Test case (TC 4.1)
- Buttonklick bei TC 1.2 verbessert

Ferias committed:
Testfall mit epocheconverter V1.5
- Webseite nach Page Objects angepasst

Ferias committed:
Dependencies angepasst

Ferias committed:
Testfall mit epocheconverter V1.6
- JavaScript-Datei in Typescript-Datei umgewandelt

Ferias committed:
Testfall mit epocheconverter V1.6 - JavaScript-Datei in Typescript-Datei umgewandelt

BenediktSt commented on angular/spec.js:
```diff
@@ -0,0 +1,149 @@
+"use strict";
```
js-Dateien sollten jetzt nicht mehr eingechecked werden, diese werden ja generiert

BenediktSt replied:
das ist noch nicht resolved

Ferias replied:
Ist entfernt

BenediktSt replied:
nein sind sie nicht

BenediktSt commented on angular/spec_convert.ts:
```diff
@@ -0,0 +1,19 @@
+import {element, by} from 'protractor';
```
naming der Datei: nur testfälle sollten auch specs gennant werden.
Page Objects am besten in einem eigenen Ordner

BenediktSt commented on angular/spec_convert.ts:
```diff
@@ -0,0 +1,19 @@
+import {element, by} from 'protractor';
+
+module.exports = {
```
Hier am besten auch mit Klassen und dann mit Attributen und Methoden arbeiten

BenediktSt commented:
Der Angular Ordner sollte immer noch entfernt werden

BenediktSt commented on angular/package.json:
```diff
@@ -0,0 +1,28 @@
+  "version": "0.0.0",
+  "license": "MIT",
+  "scripts": {
+    "ng": "ng",
```
deine scipts basieren alle auf ng = angular-cli
das ist aber nicht mehre in deinen dependencies und du brauchst das auch nicht.
Also kann das weg

BenediktSt commented on angular/package.json:
```diff
@@ -0,0 +1,28 @@
+    "build": "ng build --prod",
+    "test": "ng test",
+    "lint": "ng lint",
+    "e2e": "ng e2e"
```
neues Testing script ist tauch notwendig

BenediktSt commented on angular/protractor.conf.js:
```diff
@@ -0,0 +1,14 @@
+	'spec_timestamp',
+	'spec_ymdhms_timestamp'
+  ]
+};
```
ok also entweder
* Config so anpassen, dass man die tests mit einem befehl ohne vorherigen selenium oder so starten kann
* direct connect verwenden
* local selenium verwenden
* readme schreiben, damit man weiß, was man alles machen muss, damit man die Tests ausführen kann

BenediktSt replied:
das ist noch nicht resolved

Ferias replied:
Reicht allein directconnect nicht? Dann wird Selenium auf local ausgeführt und seleniumAddress & seleniumServerJar werden nicht berücksichtigt.

Ferias replied:
@BenediktSt, welche Erwartung hast du bei Readme?

BenediktSt replied:
directconnect reicht für den Anfang aus

BenediktSt commented on angular/protractor.conf.js:
```diff
@@ -0,0 +1,14 @@
+  specs: [
+	'spec.ts',
+	'spec_convert.ts',
+	'spec_convert_sdhm',
```
die page Objects sind keine Specs

BenediktSt commented on angular/package.json:
```diff
@@ -0,0 +1,28 @@
+    "test": "ng test",
+    "lint": "ng lint",
+    "e2e": "ng e2e"
+  },
```
ein script für den typescriptcompiler wäre auch gut, eventuell auch der webdriver-manager

BenediktSt commented on angular/spec.ts:
```diff
@@ -0,0 +1,88 @@
+import { browser, element, by } from 'protractor';
+
+let ConvertYM = require('./spec_convert.ts');
```
das sind nicht die korrekten Imports, das sollte auch dazu führen, dass deine Dateien falsch geladen werden und dann noch in Typescript referenziert werden.
Import Statements verwenden

BenediktSt commented on angular/tsconfig.json:
```diff
@@ -0,0 +1,11 @@
+    "noImplicitAny": false,
+	"lib": ["es2015"]
+  }
+}
```
ich würde hier auch einen dedizieten output order angeben, dann lässt sich das auch leichter aus git exkludieren

BenediktSt replied:
nicht resolved

Ferias replied:
Ist hingefügt woden

BenediktSt replied:
nein

Ferias committed:
PR angepasst

Ferias committed:
PR angepasst

zzz_delete_js.cmd löscht alle Dateien mit Endung .js ausgenommen protractor_conf.js

zzz_tsc.cmd führt Befehl tsc mit allen Dateien mit Endung .ts aus

Ferias committed:
Revert "Testfall mit epocheconverter V1.6 - JavaScript-Datei in Typescript-Datei umgewandelt"

This reverts commit 82fb02d4

Ferias committed:
PR angepasst

BenediktSt commented:
keine js Dateien mehr einchecken, diese sollen immer generiert werden

BenediktSt commented on package.json:
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "chris-package",
```
anderer name wäre passender

Ferias replied:
Ist geändert

Ferias committed:
Probleme behoben

Ferias committed:
Funktion bei timestamp.ts eingebaut

BenediktSt commented:
JavaScript Dateien aus der Versionsverwaltung entfernen!

BenediktSt commented on epochconverter/convert.ts:
```diff
@@ -0,0 +1,15 @@
+	public convert_month: ElementFinder = element(by.id("br-d2")).element(by.className("dateform month"));
+	public convert_button: ElementFinder = element(by.buttonText("Convert"))
+
+	constructor(year:string, month:string){
```
Keine Logik im Constructor -> Auslagern in eigene Funktion

Ferias replied:
Ist entfernt worden

BenediktSt commented on epochconverter/convert_sdhm.ts:
```diff
@@ -0,0 +1,12 @@
+	public convert_sdhm: ElementFinder = element(by.id("tc")).element(by.id("scinput"));
+	public convert_sdhm_button: ElementFinder = element(by.id("tc")).element(by.buttonText("Seconds to days, hours, minutes"));
+
+	constructor(seconds:string){
```
Keine Logik im Constructor -> Auslagern in eigene Funktion

Ferias replied:
Ist entfernt worden

BenediktSt commented on epochconverter/convert.ts:
```diff
@@ -0,0 +1,15 @@
+import {element, by, ElementFinder} from 'protractor';
+
+export class YM {
```
Sprechende Namen verwenden

Ferias replied:
Wird in nächsten Commit zu sehen sein

BenediktSt commented on epochconverter/convert_sdhm.ts:
```diff
@@ -0,0 +1,12 @@
+import {element, by, ElementFinder} from 'protractor';
+
+export class SEC {
```
Sprechende Namen verwenden

Ferias replied:
Wird in nächsten Commit zu sehen sein

BenediktSt commented on epochconverter/timestamp.ts:
```diff
@@ -0,0 +1,14 @@
+import {element, by, ElementFinder} from 'protractor';
+
+export class DTG {
```
Sprechende Namen verwenden

Ferias replied:
Wird in nächsten Commit zu sehen sein

BenediktSt commented on epochconverter/timestamp.ts:
```diff
@@ -0,0 +1,14 @@
+export class DTG {
+	public timestamp: ElementFinder = element(by.id("br-d1")).element(by.className("dateform year"));
+	public timestamp_button: ElementFinder = element(by.id("br-d2")).element(by.className("dateform month"));
+
```
Constructor fehlt -> Locator in Constructor verlagern

BenediktSt commented on epochconverter/timestamp.ts:
```diff
@@ -0,0 +1,14 @@
+	public timestamp: ElementFinder = element(by.id("br-d1")).element(by.className("dateform year"));
+	public timestamp_button: ElementFinder = element(by.id("br-d2")).element(by.className("dateform month"));
+
+	DTG_manipulate(this: DTG) {
```
was soll diese Funktion machen?
Wieso bekommt sie eine Referenz des eigenen Objekts?
Warum gibt sie eine Funktion zurück?

BenediktSt commented on epochconverter/ymdhms_timestamp.ts:
```diff
@@ -0,0 +1,27 @@
+import {element, by, ElementFinder} from 'protractor';	
+
+export class YMDHMS {
```
Sprechende Namen verwenden

BenediktSt commented on epochconverter/ymdhms_timestamp.ts:
```diff
@@ -0,0 +1,27 @@
+	public ymdhms_timestamp_seconds: ElementFinder = element(by.name("ss"));
+	public ymdhms_timestamp_button: ElementFinder = element.all(by.buttonText("Human date to Timestamp")).first();
+	
+	constructor(year: string, month: string, day: string, hours: string, minutes: string, seconds: string) {
```
Keine Logik im Constructor -> Auslagern in eigene Funktion

BenediktSt commented on spec.ts:
```diff
@@ -0,0 +1,83 @@
+	let seconds = moment().format('s');
+
+	// letiables for test 
+	let ymdhms_timestamp_year = element(by.id("hf-d1")).element(by.className("dateform year"));
```
keine Locators mehr in den tests, diese sollten in den Page Objects definiert werden

Ferias replied:
Diese Variablen braucht man für expect toEqual, ist alle notige Variablen, den Rest habe ich schon entfernt.

BenediktSt replied:
genau diese Sachen sollen aber in die Page Objects ausgelagert werden. Aktuell pflegst du diese Werte 2 Mal und hast dadurch deinen Wartungsaufwand erhöht, anstatt verringert. Unbedingt umbauen!

BenediktSt commented on spec.ts:
```diff
@@ -0,0 +1,83 @@
+
+	// TODO Ausgabe muss noch ausgelesen werden
+	it('TC 2.1 manipulate date & time and check', async function() {
+		expect(timestamp.getAttribute('value')).toEqual('Wed, 08 May 2019 15:16:02 GMT'); // It will be always wrong, just ignore this error.
```
?

Ferias replied:
Ist für mich, weil ich später noch diese Werte in Variablen umschreiben will und damit testen soll.

BenediktSt commented on spec.ts:
```diff
@@ -0,0 +1,83 @@
+    });
+
+	// DONE!
+	it('TC 1.1. Check date & time', () => {
```
Testfallname unklar -> Bessere Benennung, was wirklich getestet wird

Ferias replied:
in Check current date & time umgenannt, es soll aktuelle Datum und Uhrzeit testen.

BenediktSt commented on zzz_delete_js.cmd:
```diff
@@ -0,0 +1 @@
+del /s "C:\Users\A420160\Documents\docker\e2e-testing\epochconverter\*.js" & del /s "C:\Users\A420160\Documents\docker\e2e-testing\spec.js"
```
CMDs löschen

Ferias replied:
ist gelöscht

BenediktSt commented on zzz_tsc.cmd:
```diff
@@ -0,0 +1,2 @@
+tsc --target es5 --lib dom,es6 spec.ts & tsc --target es5 --lib dom,es6 epochconverter/convert.ts & tsc --target es5 --lib dom,es6 epochconverter/convert_sdhm.ts & tsc --target es5 --lib dom,es6 epochconverter/timestamp.ts & tsc --target es5 --lib dom,es6 epochconverter/ymdhms_timestamp.ts
```
CMDs löschen

Ferias replied:
ist gelöscht

Ferias committed:
Lösung für Webdriver Manager

Ferias committed:
Testfall mit epochconverter V1.7
- Async Funktion in TS-Dateien eingebaut
- Protractor.conf.js mit Directconnect und Output erweitert
- package.json Package-Name geändert
- cmd-Dateien entfernt
- Readme erweitert

Ferias committed:
PR angepasst

BenediktSt commented on README.md:
```diff
@@ -1,2 +1,6 @@
 # e2e-testing
-Repository für die e2e Tests von verschiedenen Applikationen
+Repository für die e2e Tests von verschiedenen Applikationen<br><br>
```
keine <br>Tags in Markdown notwendig

BenediktSt commented on epochconverter/convert.ts:
```diff
@@ -0,0 +1,15 @@
+
+	public async perform(year:string, month:string){
+		await this.convert_year.clear();
+		this.convert_year.sendKeys(year);
```
warum kein await bei sendKeys?

Ferias replied:
Done

Ferias committed:
Commit last version

Ferias committed:
Delete convert.js

Ferias committed:
Delete ymdhms_timestamp.js

BenediktSt commented on epochconverter/ymdhms_timestamp.ts:
```diff
@@ -0,0 +1,27 @@
+	public ymdhms_timestamp_seconds: ElementFinder = element(by.name("ss"));
+	public ymdhms_timestamp_button: ElementFinder = element.all(by.buttonText("Human date to Timestamp")).first();
+	
+	public async perform(year: string, month: string, day: string, hours: string, minutes: string, seconds: string) {
```
Kein sprechender Funktionsname

Ferias committed:
- Promise bei Datei ymdhms_timestamp.ts eingebaut
- Erweiterung bei .gitignore
